run_reg_test.py

```diff
@@ -0,0 +1,362 @@
+        for header in results.readline().split():
+            if header in data:
+                positions[header] = position
+            position += 1
+
+        table = PrettyTable()
+        table.field_names = list(data.keys())
+        for line in results.readlines():
+            info = line.split()
+            row = []
+            for key, values in data.items():
+                if len(info) - 1 < positions[key]:
+                    row += [""]
+                else:
+                    if values[1] == "{}" or not info[positions[key]].isnumeric():
+                        row += [("{}".format(info[positions[key]])) + values[0]]
+                    else:
+                        row += [(values[1] % float(info[positions[key]])) + values[0]]
+            table.add_row(row)
+        print(table)
+    return 0
+
+
+def run_odin_test(args, test_name):
+    """ Run ODIN II test with given test name """
+    odin_reg_script = [
+        find_vtr_file("verify_odin.sh"),
+        "--clean",
+        "-C",
+        find_vtr_file("output_on_error.conf"),
+        "--nb_of_process",
+        str(args.j),
+        "--test",
+        "{}/ODIN_II/regression_test/benchmark/".format(find_vtr_root()),
+    ]
+    if test_name == "odin_reg_full":
+        odin_reg_script[-1] += "suite/full_suite"
+    elif test_name == "odin_reg_syntax":
+        odin_reg_script[-1] += "task/syntax"
+    elif test_name == "odin_reg_arch":
+        odin_reg_script[-1] += "task/arch_sweep"
+    elif test_name == "odin_reg_operators":
+        odin_reg_script[-1] += "task/operators"
+    elif test_name == "odin_reg_large":
+        odin_reg_script[-1] += "task/large"
+    elif test_name == "odin_reg":
+        odin_reg_script[-1] += "task/full"
+    elif test_name == "odin_reg_micro":
+        odin_reg_script[-1] += "suite/light_suite"
+    else:
+        raise IOError("Test does not exist: {}".format(test_name))
+
+    odin_root = str(Path(odin_reg_script[0]).resolve().parent)
+
+    result = subprocess.call(odin_reg_script, cwd=odin_root)
+
+    assert result is not None
+    if result != 0:
+        # Error
+        print("FAILED test '{}'".format(test_name))
+        return 1
+
+    # Pass
+    print("PASSED test '{}'".format(test_name))
+    return 0
+
+
+def collect_task_list(reg_test):
+    """ create a list of task files """
+    task_list_filepath = (
+        Path(find_vtr_root())
+        / "vtr_flow"
+        / "tasks"
+        / "regression_tests"
+        / reg_test
+        / "task_list.txt"
+    )
+    if not task_list_filepath.is_file():
+        raise IOError("Test does not exist: {}".format(reg_test))
+    return str(task_list_filepath)
+
+
+def run_tasks(args, task_lists):
+    """Call 'run_vtr_task' with all the required arguments in the command"""
+    print("Running {}".format(args.reg_test[0]))
+    print(
+        "-------------------------------------------------------------------------------"
```

Comment:
Here's another divider with a different width. It would be nice to either have these be the same width, or add a function to print some text with an underline.

Reply:
This has been fixed.